subsys/settings/src/settings_its.c

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) 2019 Laczen
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ * Copyright (c) 2025 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <psa/internal_trusted_storage.h>
+#include <zephyr/settings/settings.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/psa/key_ids.h>
+
+LOG_MODULE_DECLARE(settings, CONFIG_SETTINGS_LOG_LEVEL);
+
+K_MUTEX_DEFINE(worker_mutex);
+static struct k_work_delayable worker;
+
+struct setting_entry {
+	char name[SETTINGS_MAX_NAME_LEN];
+	char value[SETTINGS_MAX_VAL_LEN];
+	size_t val_len;
+};
+
+static struct setting_entry entries[CONFIG_SETTINGS_TFM_ITS_NUM_ENTRIES];
+static int entries_count;
+
+static int settings_its_load(struct settings_store *cs, const struct settings_load_arg *arg);
+static int settings_its_save(struct settings_store *cs, const char *name, const char *value,
+			     size_t val_len);
+
+static const struct settings_store_itf settings_its_itf = {
+	.csi_load = settings_its_load,
+	.csi_save = settings_its_save,
+};
+
+static struct settings_store default_settings_its = {.cs_itf = &settings_its_itf};
+
+static int increment_uid(psa_storage_uid_t *uid)
+{
+	if (*uid < ZEPHYR_PSA_TFM_ITS_KEY_ID_RANGE_BEGIN ||
+	    *uid >= (ZEPHYR_PSA_TFM_ITS_KEY_ID_RANGE_BEGIN +
+		     ZEPHYR_PSA_TFM_ITS_KEY_ID_RANGE_SIZE)) {
```

Comment:
IMO you are missing a `-1` here because you want to ensure that `*uid` is within the valid range also at the end of the function when you will have incremented it by 1.